src/js/elements/subindicator_filter/filter_controller.js

```diff
@@ -0,0 +1,223 @@
+import {FilterRow} from "./filter_row";
+import {Observable} from "../../utils";
+import {AddFilterButton} from "../mapchip/add_filter_button";
+
+const filterRowClass = '.map-options__filter-row';
+
+class FilterControllerModel extends Observable {
+    static EVENTS = {
+        'filterRemoved': 'FilterControllerModel.filterRemoved',
+        'dataFilterModelUpdated': 'FilterControllerModel.dataFilterModelUpdated'
+    }
+
+    constructor(dataModelFilter = null) {
+        super()
+        this._filterRows = [];
+        this._dataFilterModel = dataModelFilter
+    }
+
+    get filterRows() {
+        return this._filterRows;
+    }
+
+    get dataFilterModel() {
+        return this._dataFilterModel;
+    }
+
+    removeFilterRow(filterRow) {
+        this._filterRows = this._filterRows.filter(filter => filter != filterRow)
+        this.triggerEvent(FilterControllerModel.EVENTS.filterRemoved, filterRow)
+    }
+
+    reset() {
+        this._filterRows.forEach(filterRow => {
+            this.triggerEvent(FilterControllerModel.EVENTS.filterRemoved, filterRow)
+        })
+
+        this._filterRows = []
+    }
+
+    setDataFilterModel(dataFilterModel) {
+        this.reset();
+
+        this._dataFilterModel = dataFilterModel;
+        this.triggerEvent(FilterControllerModel.EVENTS.dataFilterModelUpdated, this);
+    }
+}
+
+export class FilterController extends Observable {
+
+    static EVENTS = {
+        'ready': 'FilterController.ready'
+    }
+
+    /*
+    Note to self
+
+    About to figure out how to filter the data appropriately and send it via signal for the choropleth to be created
+    After that, evaluate how easy it is to adapt this to the charts
+
+    Things that don't currently work:
+    1. Disabling the add filter button isn't working great - it checks the number of available filters, not the number of filterRowson the screen
+    2. Removing filterRows is a little messsy
+    */
+
+    constructor(container) {
+        super();
+
+        this._container = container;
+        this._model = new FilterControllerModel();
+        this._addFilterButton = new AddFilterButton(); // TODO should pass in a container
```

Comment:
once you change AddFilterButton to extend component, you will need to pass `this` as an argument

Reply:
done